cinema/views.py

```diff
@@ -112,6 +162,24 @@ def get_serializer_class(self):
 
         return MovieSessionSerializer
 
+    def get_permissions(self):
+        if self.action in [
+            "create",
+            "list",
+            "retrieve",
+            "update",
+            "partial_update",
+            "destroy",
+        ]:
+            permission_classes = [
+                ReadAndCreate,
+            ]
+        else:
+            permission_classes = [
+                NoPermission,
+            ]
+        return [permission() for permission in permission_classes]
```

Comment:
The `get_permissions` method in `MovieSessionViewSet` applies `ReadAndCreate` permission to actions like 'update', 'partial_update', and 'destroy'. Consider using a more restrictive permission for these actions if the intention is to limit them to certain users.